File: Python/Program/Demon.py
from Karakter import Karakter

class Demon(Karakter):
    def __init__(self, nama, tinggi, umur, status, sihir, mana):
        Karakter.__init__(self, nama, tinggi, umur, status)  # langsung ke Karakter
        self._sihir = sihir   # Jenis sihir Demon
        self._mana = mana     # Jumlah mana Demon

    # Setter & Getter
    def set_sihir(self, s): self._sihir = s
    def set_mana(self, m): self._mana = m
    def get_sihir(self): return self._sihir
    def get_mana(self): return self._mana

    # Menampilkan info Demon
    def tampilkan_info(self):
        print("[Demon]")
        Karakter.tampilkan_info(self)
        print(f"Sihir: {self._sihir}, Mana: {self._mana}")
        print("---------------------------")

File: CPP/Program/Demon.cpp
#pragma once
#include "Karakter.cpp"

class Demon : virtual public Karakter {
protected:
    string Sihir; // Jenis sihir yang dimiliki Demon
    int Mana;     // Jumlah mana Demon
public:
    // Konstruktor Demon
    Demon(string nama, int tinggi, int umur, Status s, string sihir, int mana)
        : Karakter(nama, tinggi, umur, s), Sihir(sihir), Mana(mana) {}

    // Setter & Getter untuk Sihir dan Mana
    void setSihir(string s) { Sihir = s; }
    void setMana(int m) { Mana = m; }
    string getSihir() { return Sihir; }
    int getMana() { return Mana; }

    // Override tampilkanInfo untuk menampilkan info Demon
    void tampilkanInfo() override {
        cout << "[Demon]" << endl;
        Karakter::tampilkanInfo();
        cout << "Sihir: " << Sihir
             << ", Mana: " << Mana << endl;
        cout << "---------------------------" << endl;
    }
};
